x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/cache/SparseFileTracker.java

```diff
@@ -48,6 +48,12 @@ public long getLength() {
         return length;
     }
 
+    public long getPhysicalLength() {
```

Comment:
As you point it's a tricky situation. I added this method since the file systems that support `SPARSE` files, report only the written data. Meaning that during testing we check the physical length of files at disk and when we compare it with the `RecoveryState` information we get a mismatch if we use the file length. Maybe we should just rely on the OS to report the cache file physical size using the file descriptor available at `CacheFile`? wdyt @tlrx